tfjs-converter/src/executor/graph_executor.ts

```diff
@@ -414,23 +412,31 @@ export class GraphExecutor implements FunctionExecutor {
     // Graph with control flow op requires runtime evaluation of the execution
     // order, while without control flow the execution order is pre-determined
     // in the compile method.
-    this.tensorsMap = await this.executeWithControlFlow(
+    const tensorsMap = await this.executeWithControlFlow(
         inputs, context, outputs, isFunctionExecution);
-    const results =
-        outputs.map(name => getTensor(name, this.tensorsMap, context));
+    const results = outputs.map(name => getTensor(name, tensorsMap, context));
 
     // dispose all the intermediate tensors
     const outputIds = results.map(t => t.id);
     const inputIds = Object.keys(inputs).map(name => inputs[name].id);
-    this.keepIds =
+    const keepIds =
         new Set<number>([...outputIds, ...inputIds, ...this.weightIds]);
-    if (!this.keepTensorForDebug) {
-      this.disposeTensorsMap();
+    if (this.keepIntermediateTensors) {
+      this.clonedTensorsMap = this.cloneTensorMap(tensorsMap);
     }
 
+    Object.entries(tensorsMap).forEach(([, tensorsList]) => {
+      tensorsList.forEach(tensor => {
+        if (tensor && !tensor.kept && !tensor.isDisposed &&
```

Comment:
In fact I am not clear why kept is checked here, this is from the original version: https://github.com/tensorflow/tfjs/pull/5659/files#diff-44e0a825cd7c6f31c03d9333db5dc21a8937d66a5b28d719c699863eef96ad8dL369.

I just try to not change the code I am not clear.

Reply:
My guess is this was from the previous logic that used `kept` to save the intermediate tensors, although I could be wrong.

It looks like the `keepIds` set is what prevents this from disposing the input, output, and weight tensors, so we're not using `tensor.kept` to prevent them from being disposed. If all the other tensors in `tensorsMap` are intermediate tensors, then I think it's safe to remove this check.

Reply:
removed